### ISSUE TITLE
Lint/UselessMethodDefinition false positive on methods transformed by class method

### ISSUE THREAD

**Issue by Thibaut** [rubocop/rubocop#11704]
## Expected behavior

The following code should not get picked up / auto-corrected by the `UselessMethodDefinition` cop:
```ruby
memoize def my_method
  super
end
```
(illustrated here in conjunction with [Memery](https://github.com/tycooon/memery), but this applies more generally)

## Actual behavior

The code gets auto-corrected to the following invalid code:

```ruby
memoize
```
